modules/light-clients/08-wasm/types/contract_api.go

```diff
@@ -0,0 +1,121 @@
+package types
+
+import (
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+)
+
+// instantiateMessage
+type instantiateMessage struct {
+	ClientState    *ClientState    `json:"client_state"`
+	ConsensusState *ConsensusState `json:"consensus_state"`
+}
+
+// queryMsg is used to encode query messages
+// omitempty tag is mandatory for JSON serialization
+// to be compatible with Rust contract enum matching
+type queryMsg struct {
+	Status               *statusMsg               `json:"status,omitempty"`
+	ExportMetadata       *exportMetadataMsg       `json:"export_metadata,omitempty"`
+	TimestampAtHeight    *timestampAtHeightMsg    `json:"timestamp_at_height,omitempty"`
+	VerifyClientMessage  *verifyClientMessageMsg  `json:"verify_client_message,omitempty"`
+	VerifyMembership     *verifyMembershipMsg     `json:"verify_membership,omitempty"`
+	VerifyNonMembership  *verifyNonMembershipMsg  `json:"verify_non_membership,omitempty"`
+	CheckForMisbehaviour *checkForMisbehaviourMsg `json:"check_for_misbehaviour,omitempty"`
+}
+
+type statusMsg struct{}
+type exportMetadataMsg struct{}
+type timestampAtHeightMsg struct {
+	Height exported.Height `json:"height"`
+}
```

Comment:
gofumpt disagrees with you guys, @damiannolan and @chatton, and wants to put here a type block. I respect your opinion a lot, guys, you know that, but my religion doesn't allow me to break the 11th commandment: "You will never fight against your beloved linter". :)